server/src/main/java/org/opensearch/index/codec/composite/composite99/Composite99DocValuesWriter.java

```diff
@@ -135,9 +239,9 @@ private void createCompositeIndicesIfPossible(DocValuesProducer valuesProducer,
         // we have all the required fields to build composite fields
         if (compositeFieldSet.isEmpty()) {
             for (CompositeMappedFieldType mappedType : compositeMappedFieldTypes) {
-                if (mappedType.getCompositeIndexType().equals(CompositeMappedFieldType.CompositeFieldType.STAR_TREE)) {
+                if (mappedType instanceof StarTreeMapper.StarTreeFieldType) {
                     try (StarTreesBuilder starTreesBuilder = new StarTreesBuilder(state, mapperService)) {
-                        starTreesBuilder.build(fieldProducerMap);
+                        starTreesBuilder.build(metaOut, dataOut, fieldProducerMap, composite99DocValuesConsumer);
                     }
```

Comment:
Don't we have any multi star tree fields tests to catch this ?

Reply:
not sure what you mean by catch this?

Reply:
Multi startree tests are directly reaching basestartree builder. We can add more ITs around multi star tree tests